full-service/src/config.rs

```diff
@@ -351,17 +351,29 @@ impl APIConfig {
             .error_for_status()?;
         let data = response.text()?;
         let data_json: serde_json::Value = serde_json::from_str(&data)?;
-        if let Some(v) = data_json.get("country") {
-            if let Some(country) = v.as_str() {
-                match country {
-                    "US" => Err(ConfigError::InvalidCountry),
-                    _ => Ok(()),
-                }
-            } else {
-                Err(ConfigError::DataMissing(data_json.to_string()))
-            }
-        } else {
-            Err(ConfigError::DataMissing(data_json.to_string()))
+
+        let data_missing_err = Err(ConfigError::DataMissing(data_json.to_string()));
+        let country: &str = match data_json["country"].as_str() {
+            Some(c) => c,
+            None => return data_missing_err,
+        };
+        let region: &str = match data_json["region"].as_str() {
+            Some(r) => r,
+            None => return data_missing_err,
+        };
+
+        let err = Err(ConfigError::InvalidCountry);
+        match country {
+            "US" => err,
```

Comment:
You should be able to do `"US" | "IR" | "SY" | ...` which I think is more pleasant to look at but if clippy doesn't prefer one way or the other I don't feel strongly about it.